client/snap_op.go

```diff
@@ -20,87 +20,113 @@
 package client
 
 import (
+	"bytes"
+	"encoding/json"
 	"fmt"
+	"io"
+	"mime/multipart"
 	"os"
-	"strings"
+	"path/filepath"
+	"strconv"
 )
 
+type SnapOptions struct {
+	Channel string `json:"channel,omitempty"`
```

Comment:
It's not a big deal but this allows to sideload a snap *and* specify a channel which is meaningless. I don't mind this as I understand how it makes everything below much more streamlined.